src/box/sql/mem.c

```diff
@@ -2868,6 +2870,24 @@ port_lua_get_vdbemem(struct port *base, uint32_t *size)
 					 field.sval.len) != 0)
 				goto error;
 			break;
+		case MP_EXT: {
+			char buf[BUF_SIZE];
```

Comment:
buf is only 32 bytes. If user returns varbin containing more than 32 bytes str instance will crash on assertion. This doesn't look right

Reply:
Fix this please and add corresponding test.

Reply:
For the two available MP_EXT values, the length does not actually exceed 32 (for UUID - 18, for DECIMAL - up to 32). Although you are right, it was possible to get an error when DECIMAL was 32 because the assert said the length should be less than 32. Fixed, I replaced the static buf with the allocated one using region_alloc(). Also, I replaced the generic decimal with a decimal with the largest possible length (which is 38).